unittests/test_sanity_functions.py

```diff
@@ -511,41 +511,49 @@ def tempfile(tmp_path):
 
 
 @pytest.fixture
-def temptxt(tempfile):
+def contents(tempfile):
     with open(tempfile, 'r') as fp:
         return fp.read()
 
 
-def test_assert_found(tempfile):
-    assert sn.assert_found(r'Step: \d+', tempfile)
-    assert sn.assert_found(r'Step: \d+', sn.defer(tempfile))
-    with pytest.raises(SanityError):
-        sn.evaluate(sn.assert_found(r'foo: \d+', tempfile))
+def fixture(funcs):
+    @pytest.fixture(params=funcs)
+    def _fixture(request, tempfile, contents):
+        import functools
 
+        if request.param.__name__.endswith('_s'):
+            return request.param, contents
+        else:
+            return request.param, tempfile
 
-def test_assert_found_s(temptxt):
-    assert sn.assert_found_s(r'Step: \d+', temptxt)
-    assert sn.assert_found_s(r'Step: \d+', sn.defer(temptxt))
-    with pytest.raises(SanityError):
-        sn.evaluate(sn.assert_found_s(r'foo: \d+', temptxt))
+    return _fixture
 
 
-def test_assert_found_encoding(utf16_file):
-    assert sn.assert_found('Odyssey', utf16_file, encoding='utf-16')
+assert_found = fixture([sn.assert_found, sn.assert_found_s])
+assert_not_found = fixture([sn.assert_not_found, sn.assert_not_found_s])
+findall = fixture([sn.findall, sn.findall_s])
+extractall = fixture([sn.extractall, sn.extractall_s])
+extractsingle = fixture([sn.extractsingle, sn.extractsingle_s])
```

Comment:
```suggestion
_assert_found = fixture([sn.assert_found, sn.assert_found_s])
_assert_not_found = fixture([sn.assert_not_found, sn.assert_not_found_s])
_findall = fixture([sn.findall, sn.findall_s])
_extractall = fixture([sn.extractall, sn.extractall_s])
_extractsingle = fixture([sn.extractsingle, sn.extractsingle_s])
```

So that you can later avoid the generic names such as `assert_func`.